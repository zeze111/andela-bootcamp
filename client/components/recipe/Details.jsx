import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

import PreLoader from '../common/PreLoader';

/** Stateless component to render recipe details
 *
 * @export {function} Details
 *
 * @param {object} props
 *
 * @returns {null} null
 */
const Details = (props) => {
  const creatorUser = (

    <div className="col s12 m12 l12 creator right-align edit-text">
      <Link
        to={`/update-recipe/${props.recipe.id}`}
        href={`/update-recipe/${props.recipe.id}`}
        className="text-color col l3 m4 push-l5 push-m3"
      >
        Edit this recipe
      </Link>
      <button
        className="creator-button
        text-color
        right
        delete-text
        text-style
        white left-align
        col l3 m4"
        onClick={props.clickEvent}
      >Delete this recipe
      </button>
    </div >
  );

  const guestUser = (
    <div />
  );

  return (
    <div className="row">
      <div className="col s10 l9 m10 offset-s1 offset-l1 offset-m1">
        <h5 className="center-align title text-headers" > Recipe </h5>
        {(props.ingredients.length <= 0) &&
        <div className="center-align loader-style">
          <PreLoader />
        </div>
      }
        {(props.ingredients.length > 0) &&
        <div className="row">
          <div id="time" className="col l4 m4 s12 ">
            <p
              className="recipe"
            > Preparation Time:
            </p>
            {props.recipe.preparationTime}
          </div>
          <div id="Ing" className="col l4 m4 s12 r-ingredients">
            <p className="recipe"> Ingredients: </p>
            {props.ingredients}
          </div>
          <div id="Ins" className="col l4 m4 s12">
            <p className="recipe"> Instructions: </p>
            {props.instructions}
          </div>
          {(props.id === props.recipe.userId) ? creatorUser : guestUser}
          {props.isLoading &&
            <div className="right-align delete">
              <PreLoader />
            </div>
          }
        </div>
      }
      </div>
    </div>

  );
};

Details.propTypes = {
  clickEvent: PropTypes.func.isRequired,
  recipe: PropTypes.objectOf(PropTypes.any),
  ingredients: PropTypes.arrayOf(PropTypes.any),
  instructions: PropTypes.objectOf(PropTypes.any),
  id: PropTypes.number,
  isLoading: PropTypes.bool.isRequired
};

Details.defaultProps = {
  recipe: {},
  ingredients: [],
  instructions: {},
  id: 0
};

export default Details;
