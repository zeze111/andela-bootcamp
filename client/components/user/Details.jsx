import React from 'react';
import PropTypes from 'prop-types';

import PreLoader from '../common/PreLoader';
import { TextFieldGroup3 } from '../common/TextFieldGroup';

/** Stateless component to render user's form and details
 *
 * @export {function} Details
 *
 * @param {object} props
 *
 * @returns {null} null
 */
const Details = (props) => {
  if (props.isLoading) {
    return (
      <div className="center-align loader-style">
        <PreLoader />
      </div>
    );
  }
  return (
    <div id="detail" className="col s12 m8 l7 push-l1 push-m1 form-style">
      <form onSubmit={props.onSubmit} className="col s10 push-s1">
        <TextFieldGroup3
          label="First Name"
          value={props.firstName}
          onChange={props.onChange}
          id="ufname"
          type="text"
          name="firstName"
          icon="account_circle"
        />
        <TextFieldGroup3
          label="Surname:"
          value={props.surname}
          onChange={props.onChange}
          id="usname"
          type="text"
          name="surname"
          icon="account_circle"
        />
        <TextFieldGroup3
          label="Email:"
          value={props.email}
          onChange={props.onChange}
          id="useremail"
          type="email"
          name="email"
          icon="email"
          isDisabled
        />
        <TextFieldGroup3
          label="Bio:"
          value={props.bio}
          onChange={props.onChange}
          id="bio"
          type="text"
          name="bio"
          icon="mode_edit"
        />
        <div className="right-align">
          <button
            disabled={props.isLoading}
            className="btn grey"
            type="submit"
          > Update
          </button>
        </div >
      </form>
    </div>
  );
};

Details.defaultProps = {
  firstName: '',
  surname: '',
  email: '',
  bio: ''
};

Details.propTypes = {
  firstName: PropTypes.string,
  email: PropTypes.string,
  surname: PropTypes.string,
  bio: PropTypes.string,
  isLoading: PropTypes.bool.isRequired,
  onChange: PropTypes.func.isRequired,
  onSubmit: PropTypes.func.isRequired,
};

export default Details;
