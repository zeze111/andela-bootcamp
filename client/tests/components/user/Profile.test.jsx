import React from 'react';

import connectedProfile, {Profile} from '../../../components/user/Profile';
import Details from '../../../components/user/Details';
import store from '../../../store';

let props =  {
  getUserRecipes: jest.fn(() => Promise.resolve()),
  deleteRecipe: jest.fn(() => Promise.resolve()),
  getFavoriteRecipes: jest.fn(() => Promise.resolve()),
  deleteFavorite: jest.fn(() => Promise.resolve()),
  updateUser: jest.fn(() => Promise.resolve()),
  getUser: jest.fn(() => Promise.resolve()),
  changePassword: jest.fn(() => Promise.resolve()),
  recipes: [
    {
      id: 1,
      name: 'something'
    },
    {
      id: 2,
      name: 'something'
    },
  ],
  favorites: [
    {
      id: 1,
      name: 'something'
    },
    {
      id: 2,
      name: 'something'
    },
  ]
};

const setup = () => {
  const wrapper = shallow(<Profile {...props} store={store} />);
  return { wrapper }
};

describe('component will receive props', () => {
  it('should call get recipes function', () => {
    const spy = sinon.spy(Profile.prototype, 'componentWillReceiveProps');

    const nextProps = {
      getUserRecipes: jest.fn(() => Promise.resolve()),
      deleteRecipe: jest.fn(() => Promise.resolve()),
      getFavoriteRecipes: jest.fn(() => Promise.resolve()),
      deleteFavorite: jest.fn(() => Promise.resolve()),
      updateUser: jest.fn(() => Promise.resolve()),
      getUser: jest.fn(() => Promise.resolve()),
      changePassword: jest.fn(() => Promise.resolve()),
      message: '',
      userMessage: '',
      user: {},
      match: { params: { tab: 'some' } },
      recipes: [],
      pagination: {},
      pagination2: {},
      favorites: []
    };

    const wrapper = shallow(<Profile {...nextProps} />);
    wrapper.setProps({ prop: 15 });
    expect(spy.calledOnce).toEqual(true);
  });
});

describe('onChange', () => {
  it('should change event', () => {
    const event = {
      target: { name: 'email', value: 'Doe@yahoo.com' },
    };

    const { wrapper } = setup();
    expect(wrapper.instance().onChange(event));
    expect(wrapper.instance().state.email).toBe('Doe@yahoo.com');
  });
});

describe('onSubmit', () => {
  it('be called when form is submitted', () => {
    const { wrapper } = setup();
    const event = {
      preventDefault: jest.fn(),
    };

    wrapper.setState({
      surname: 'janedoe',
    });

    wrapper.instance().onSubmit(event);
    expect(props.updateUser.mock.calls.length).toEqual(1);
  });
});

describe('onSubmit', () => {
  it('return validation error when form is submitted', () => {
    const { wrapper } = setup();
    const event = {
      preventDefault: jest.fn(),
    };

    wrapper.setState({
      surname: 'j',
    });
    wrapper.instance().onSubmit(event);
    expect(props.updateUser.mock.calls.length).toEqual(1);
  });
});

describe('onSubmit', () => {
  it('return an error when form is submitted', () => {
    const event = {
      preventDefault: jest.fn(),
    };
    props.updateUser = jest.fn(() => Promise.reject());
    const wrapper = shallow(<Profile {...props} store={store} />);

    wrapper.setState({
      surname: 'janedo'
    });
    wrapper.instance().onSubmit(event);
    expect(props.updateUser.mock.calls.length).toEqual(1);
  });
});

describe('onNextPage', () => {
  it('is called when next/previous page is called for user\'s recipes ', () => {
    const { wrapper } = setup();
    const event = 1;

    wrapper.instance().onNextPage(event);
  });
});

describe('onNextFavePage', () => {
  it('is called when next/previous page is called for user\'s favorites', () => {
    const { wrapper } = setup();
    const event = 1;

    wrapper.instance().onNextFavePage(event);
  });
});

describe('uploadImage', () => {
  it('is called when uploads an image', () => {
    const { wrapper } = setup();
    const event = {
      target: {
        files: ['pic.bg']
      }
    };

    wrapper.instance().uploadImage(event);
    // expect(wrapper.instance().uploadImage.mock.length).toEqual(1);
  });
});

describe('Test for Profile', () => {
  it('should render component', () => {
    document.body.innerHTML =
    '<div>' +
    '  <img class="materialboxed" />' +
    '  <button id="button" />' +
    '</div>';

    const wrapper = setup();
    expect(wrapper).toMatchSnapshot()
  });
});
