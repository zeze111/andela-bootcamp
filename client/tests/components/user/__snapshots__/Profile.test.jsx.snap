// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for Profile should render component 1`] = `
Object {
  "wrapper": <div
    id="profile-body"
  >
    <main>
      <div
        className="container full-container two-down"
      >
        <br />
         
        <br />
        <Information
          image=""
          isLoading={true}
          isPicLoading={false}
          profile={
            Object {
              "email": "",
              "firstName": "",
              "surname": "",
            }
          }
          uploadImage={[Function]}
        />
      </div>
      <div
        className="container z-depth-1 white "
      >
        <div
          className="row two-down"
        >
          <div
            className="col s12 m12 l12"
          />
           
          <br />
           
          <br />
        </div>
      </div>
    </main>
  </div>,
}
`;
